bench/wtperf/wtperf.c

```diff
@@ -456,6 +475,46 @@ worker(void *arg)
 		goto err;
 	}
 
+	/* Setup for truncate */
+	if (thread->workload->truncate != 0) {
+
+		STAILQ_INIT(&truncate_stone_head);
+
+		/* Truncation % value. eg 10% is 0.1 */
+		truncation_percentage =
+		    (double)thread->workload->truncate_pct / 100;
+		/* How wide each milestone is */
+		truncate_milestone_gap =
+		    thread->workload->truncate_count*truncation_percentage;
+		/* How many milestones we need */
+		needed_milestones = thread->workload->truncate_count /
+		    truncate_milestone_gap;
+
+		/* Setup the initial milestones if we have data */
+		if (cfg->insert_key != 0) {
+			starting_point = 0;
+			/* Shouldn't happen */
+			if (cfg->insert_key > thread->workload->truncate_count)
```

Comment:
I think this would be better as something like (pseudo code):

```
start_point = cursor->next();
start_point_key = copy_key(cursor->get_key());
start_point_val = decode_key_from_string(start_point_key);
cursor->reset();
end_point = cursor->prev();
end_point_key = copy_key(cursor->get_key());
start_point_val = decode_key_from_string(end_point_key);

initial_milestone_gap = (end_point_val - start_point_val) / needed_milestones;
/* Deal with pathological case where there is less content than milestones. */
if (initial_milestone_gap == 0)
    initial_milestone_gap = 1;
```